pkg/storage/vault/aws/aws.go

```diff
@@ -0,0 +1,135 @@
+package aws
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"strconv"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	awsConfig "github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/secretsmanager"
+	"github.com/scratchdata/scratchdata/config"
+)
+
+type AWSVault struct {
+	client *secretsmanager.Client
+	prefix string
+}
+
+func NewAWSVault(conf map[string]any, destinations []config.Destination) (*AWSVault, error) {
+	cfg, err := awsConfig.LoadDefaultConfig(context.Background())
+	if err != nil {
+		return nil, err
+	}
+
+	var accessKey, secretKey, prefix, region string
+
+	if accessKeyVal, ok := conf["AccessKey"].(string); ok {
```

Comment:
All of these configurations should be part of the `AWSVault` struct.

Reply:
Please use the exact same pattern as you used [here](https://github.com/scratchdata/scratchdata/pull/120/files#diff-39009ce4deb7c44edd1c180471d6d79ea36e471a012dd4725443057dbdb94870R14) to automatically map configuration parameters from the yaml file into the struct.

Reply:
Please use the same names of the aws yaml configuration key as we're using elsewhere in the code. https://github.com/scratchdata/scratchdata/blob/main/pkg/storage/queue/sqs/sqs.go#L17-L19